modules/javafx.web/src/main/java/com/sun/javafx/webkit/prism/WCGraphicsPrismContext.java

```diff
@@ -523,6 +528,55 @@ public void fillRect(final float x, final float y, final float w, final float h,
                     final NGRectangle node = new NGRectangle();
                     node.updateRectangle(x, y, w, h, 0, 0);
                     render(g, shadow, paint, null, node);
+                } else if(state.getClipMaskImageNoClone() != null) {
+                    Rectangle rect = new Rectangle((int) x, (int) y, (int) w, (int) h);
+                    Rectangle transformedRect = new Rectangle();
+                    state.getTransformNoClone().transform(rect, transformedRect);
+                    RTTexture paintRtTexture = g.getResourceFactory().createRTTexture(
+                            (int) Math.ceil(transformedRect.width),
+                            (int) Math.ceil(transformedRect.height),
+                            Texture.WrapMode.CLAMP_NOT_NEEDED);
+                    Graphics g1 = paintRtTexture.createGraphics();
+                    g1.setPaint(paint);
+                    g1.setTransform(PrismGraphicsManager.getPixelScaleTransform());
+                    g1.fillRect(0, 0, paintRtTexture.getContentWidth(), paintRtTexture.getContentHeight());
+
+                    RTImage maskImage = (RTImage) (state.getClipMaskImageNoClone());
+                    Image nativeMaskImage = Image.fromByteBgraPreData(maskImage.getPixelBuffer(), maskImage.getWidth(),
+                            maskImage.getHeight(), 0, maskImage.getPixelScale());
+                    Texture maskTexture = g.getResourceFactory().createTexture(nativeMaskImage, Texture.Usage.STATIC,
+                            Texture.WrapMode.CLAMP_NOT_NEEDED);
+                    RTTexture maskRtTexture = g.getResourceFactory().createRTTexture(nativeMaskImage.getWidth(),
+                            nativeMaskImage.getHeight(), Texture.WrapMode.CLAMP_NOT_NEEDED);
+                    Graphics g2 = maskRtTexture.createGraphics();
+                    g2.setTransform(PrismGraphicsManager.getPixelScaleTransform());
+                    g2.drawTexture(maskTexture, 0, 0, nativeMaskImage.getWidth(), nativeMaskImage.getHeight());
+                    maskTexture.dispose();
+                    if(g instanceof MaskTextureGraphics && !(g instanceof PrinterGraphics)) {
+                        MaskTextureGraphics mg = (MaskTextureGraphics) (g);
+                        mg.drawPixelsMasked(paintRtTexture, maskRtTexture, transformedRect.x, transformedRect.y,
+                                transformedRect.width, transformedRect.height, 0, 0, 0, 0);
+                    } else {
+                        Screen screen = g.getAssociatedScreen();
```

Comment:
This logic is already present in getFilterContext(). You can call the function instead.